### PR TITLE
🐛 Fix ESM sourcemaps

### PR DESCRIPTION
**Pull request by jridgewell**
In regular builds, the v0.js file starts with:

```js
var global=self;self.AMP=self.AMP||[];try{(function(_){
```

That's it. It's an effectively "empty" line, and we can freely prepend code to this line without effecting our sourcemaps. As long as the prepended code never includes a newline, the sourcemap will fix itself on the next line.

However in ESM builds, we run terser. Terser will concat lines together, making the first line:

```js
var global=self;self.AMP=self.AMP||[];try{(function(t){var e=class{constructor(){this.Ff=new i}abort(){this.Ff.df=!0}get signal(){return this.Ff}};…snip 11kb…
```

When we prepended our config onto this line, we affected the sourcemap for all of the later code on the line! A simple fix is to just force terser to prepend a newline before any code, so we're again free to prepend our config without direct sourcemap awareness.

### CODE REVIEW

**Review by rcebulko — APPROVED** [review]
Thanks for fixing this so quickly

**Comment by rcebulko** [comment]
This may be worth a cherry-pick into Beta/Experimental to rescue the next week of error reporting

File: build-system/tasks/check-sourcemaps.js [hunk 1]
```diff
@@ -179,10 +185,8 @@ function checkSourcemapMappings(sourcemapJson) {
  */
 async function checkSourcemaps() {
   maybeBuild();
-  const sourcemapJson = getSourcemapJson();
-  checkSourcemapUrl(sourcemapJson);
-  checkSourcemapSources(sourcemapJson);
-  checkSourcemapMappings(sourcemapJson);
+  checkSourceMap(v0JsMap);
+  checkSourceMap(v0MjsMap);
```
Comment (samouri) [inline build-system/tasks/check-sourcemaps.js:189]: Thanks for updating the tests!

File: build-system/tasks/check-sourcemaps.js [hunk 2]
```diff
@@ -165,12 +167,16 @@ function checkSourcemapMappings(sourcemapJson) {
     log(helpMessage);
     throwError('Found mapping for incorrect code');
   }
-  if (generatedCodeColumn != expectedFirstLineColumn || sourceCodeColumn != 0) {
```
Comment (samouri) [inline build-system/tasks/check-sourcemaps.js:168]: Why are we deleting the column check?

Comment (jridgewell) [reply build-system/tasks/check-sourcemaps.js:168]: We're already testing `firstLineFile === expectedFirstLineFile`, which is the unmapping part we care about. This `generatedCodeColumn` doesn't actually matter (we only care that it's the first mapping segment and points to the correct spot). Between ESM and regular builds, the actual generated column is different (`"use strict"` isn't necessary in ESM).

**Review by samouri — APPROVED** [review]

File: build-system/tasks/extension-helpers.js [hunk 3]
```diff
@@ -178,6 +178,9 @@ function getExtensionsToBuild(preBuild = false) {
   if (extensionsToBuild) {
     return extensionsToBuild;
   }
+  if (argv.core_runtime_only) {
+    return (extensionsToBuild = []);
+  }
```
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:183]: @jridgewell This change breaks the ability to run `gulp dist --core_runtime_only --extensions=amp-foo` (added in #31091), which lets you compile `v0.js` and select extension(s), but skip all the other non-`v0.js` entry points.

Comment (jridgewell) [reply build-system/tasks/extension-helpers.js:183]: I'll send an update. I was tired of `--core_runtime_only` building the default extensions.

**Pull request merged**